test/unit/mallctl.c

```diff
@@ -1223,6 +1223,27 @@ TEST_BEGIN(test_thread_activity_callback) {
 }
 TEST_END
 
+TEST_BEGIN(test_arenas_create_ext) {
+	unsigned arena, arena1;
+	void *ptr;
+	size_t sz = sizeof(unsigned);
+
+	arena_config_t config;
+	config.extent_hooks = NULL;
+	config.metadata_use_hooks = false;
+
+	expect_d_eq(mallctl("experimental.arenas_create_ext",
```

Comment:
Sorry for the extra pain, but could you also add an extra test that passes in non-default hooks as well (and, perhaps, asserts that they don't get called, or at least don't get called until a real user alloc happens)? I don't think you missed anywhere in this diff, but I can imagine it being easy to accidentally use the extent hooks for metadata in future refactors.

Reply:
No worries, you are right. I was just being lazy :) I've added a couple of new tests to verify that there are no base allocations performed during arena create ctl. I also moved those tests from `unit/mallctl.c` to `integration/extent.c` - it seemed like a more appropriate place, but I'm not sure.